homeassistant/components/tolosauna/climate.py

```diff
@@ -0,0 +1,166 @@
+
+
+class SaunaClimate(ToloSaunaCoordinatorEntity, ClimateEntity):
+    """Sauna climate control."""
+
+    def __init__(
+        self, coordinator: ToloSaunaUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize TOLO Sauna Climate entity."""
+        super().__init__(coordinator, entry)
+
+        self._attr_fan_modes = [FAN_ON, FAN_OFF]
+        self._attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_DRY]
+        self._attr_max_humidity = DEFAULT_MAX_HUMIDITY
+        self._attr_max_temp = DEFAULT_MAX_TEMP
+        self._attr_min_humidity = DEFAULT_MIN_HUMIDITY
+        self._attr_min_temp = DEFAULT_MIN_TEMP
+        self._attr_name = "Sauna Climate"
+        self._attr_precision = PRECISION_WHOLE
+        self._attr_supported_features = (
+            SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY | SUPPORT_FAN_MODE
+        )
+        self._attr_target_temperature_step = 1
+        self._attr_temperature_unit = TEMP_CELSIUS
+        self._attr_unique_id = f"{self._config_entry.entry_id}_climate"
+
+    @property
+    def current_temperature(self) -> int:
+        """Return current temperature."""
+        return self.status.current_temperature
+
+    @property
+    def current_humidity(self) -> int:
+        """Return current humidity."""
+        return self.status.current_humidity
+
+    @property
+    def target_temperature(self) -> int:
+        """Return target temperature."""
+        return self.settings.target_temperature
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return self.settings.target_humidity
+
+    @property
+    def hvac_mode(self) -> str:
+        """Get current HVAC mode."""
+        if self.status.power_on:
+            return HVAC_MODE_HEAT
+        if not self.status.power_on and self.status.fan_on:
+            return HVAC_MODE_DRY
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_action(self) -> str | None:
+        """Execute HVAC action."""
+        if self.status.calefaction == Calefaction.HEAT:
+            return CURRENT_HVAC_HEAT
+        if self.status.calefaction == Calefaction.KEEP:
+            return CURRENT_HVAC_IDLE
+        if self.status.calefaction == Calefaction.INACTIVE:
+            if self.status.fan_on:
+                return CURRENT_HVAC_DRY
+            return CURRENT_HVAC_OFF
+        return None
+
+    @property
+    def fan_mode(self) -> str:
+        """Return current fan mode."""
+        if self.status.fan_on:
+            return FAN_ON
+        return FAN_OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set HVAC mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self.hass.async_add_executor_job(
+                self._set_power_and_fan, False, False
+            )
+        if hvac_mode == HVAC_MODE_HEAT:
+            await self.hass.async_add_executor_job(self._set_power_and_fan, True, False)
```

Comment:
Instead of using `async_add_executor_job`, change this method to be sync.

`def set_hvac_mode(self, hvac_mode: str) -> None:`